libshaderc/src/shaderc_test.cc

```diff
@@ -232,6 +232,37 @@ TEST(CompileStringWithOptions, IfDefCompileOption) {
                                  shaderc_glsl_vertex_shader, options.get()));
 }
 
+TEST(CompileStringWithOptions, MessageRules) {
+  Compiler compiler;
+  compile_options_ptr options(shaderc_compile_options_initialize());
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+
+  // Confirm that this shader only compiles with shaderc_msg_default;
+  // if using SPIR-V rules, glslang will fail to compile it
+  const std::string kGlslShader =
```

Comment:
Thanks for the test.  It was really illuminating.  It made me think hard about what the option was really doing.